security_hub_accepter.py

```diff
@@ -0,0 +1,153 @@
+"""Helper script to accept invite for cross-account RAM Share."""
+from __future__ import print_function
+
+import argparse
+import collections
+import logging
+import os
+import sys
+import time
+
+import botocore.credentials
+import botocore.session
+
+DEFAULT_LOG_LEVEL = logging.INFO
+LOG_LEVELS = collections.defaultdict(
+    lambda: DEFAULT_LOG_LEVEL,
+    {
+        "critical": logging.CRITICAL,
+        "error": logging.ERROR,
+        "warning": logging.WARNING,
+        "info": logging.INFO,
+        "debug": logging.DEBUG,
+    },
+)
+
+# Clear out any other root log handler
+root = logging.getLogger()
+if root.handlers:
+    for handler in root.handlers:
+        root.removeHandler(handler)
+
+logging.basicConfig(
+    format="%(asctime)s.%(msecs)03dZ [%(name)s][%(levelname)-5s]: %(message)s",
+    datefmt="%Y-%m-%dT%H:%M:%S",
+    level=LOG_LEVELS[os.environ.get("LOG_LEVEL", "").lower()],
+)
+log = logging.getLogger("security_hub_accepter.py")
+
+
+class NoPendingInviteException(Exception):
+    pass
+
+
+class AssumeRoleProvider(object):
+    METHOD = "assume-role"
+
+    def __init__(self, fetcher):
+        self._fetcher = fetcher
+
+    def load(self):
+        return botocore.credentials.DeferredRefreshableCredentials(
+            self._fetcher.fetch_credentials, self.METHOD
+        )
+
+
+def filter_none_values(data):
+    """Returns a new dictionary excluding items where value was None"""
+    return {k: v for k, v in data.items() if v is not None}
+
+
+def assume_role(
+    session, role_arn, duration=3600, session_name=None, serial_number=None
+):
+    fetcher = botocore.credentials.AssumeRoleCredentialFetcher(
+        session.create_client,
+        session.get_credentials(),
+        role_arn,
+        extra_args=filter_none_values(
+            {
+                "DurationSeconds": duration,
+                "RoleSessionName": session_name,
+                "SerialNumber": serial_number,
+            }
+        ),
+        cache=botocore.credentials.JSONFileCache(),
+    )
+    role_session = botocore.session.Session()
+    role_session.register_component(
+        "credential_provider",
+        botocore.credentials.CredentialResolver([AssumeRoleProvider(fetcher)]),
+    )
+    return role_session
+
+
+def get_pending_invite_id(sechub, master_account_id):
+    invitations = sechub.list_invitations()
+
+    for invite in invitations["Invitations"]:
+        invite_id = invite["InvitationId"]
+        if (
+            invite["MemberStatus"] == "Invited"
+            and invite["AccountId"] == master_account_id
+        ):
+            return invite_id
+
+
+def accept_pending_invite(sechub, master_account_id, invite_id):
+    sechub.accept_invitation(MasterId=master_account_id, InvitationId=invite_id)
+
+
+def main(
+    master_account_id, remove_master=False, profile=None, role_arn=None, region=None
+):
+    region = region or None
```

Comment:
isn't this redundant, or am I missing something?

Reply:
Yeah, idk. Some copy pasta magic. Removed.

Reply:
So tried it without it and it fails so not sure why this is necessary. @lorengordon any insights?

Reply:
terraform passes the values through as empty strings. . . . this coerces them back to `None`